interfaces/builtin/io_ports_control_test.go

```diff
@@ -25,70 +25,104 @@ import (
 	"github.com/snapcore/snapd/interfaces"
 	"github.com/snapcore/snapd/interfaces/builtin"
 	"github.com/snapcore/snapd/snap"
-	"github.com/snapcore/snapd/testutil"
+	"github.com/snapcore/snapd/snap/snaptest"
 )
 
-type IoPortsControlInterfaceSuite struct {
+type IioPortsControlInterfaceSuite struct {
 	iface interfaces.Interface
 	slot  *interfaces.Slot
 	plug  *interfaces.Plug
 }
 
-var _ = Suite(&IoPortsControlInterfaceSuite{
-	iface: builtin.NewIoPortsControlInterface(),
-	slot: &interfaces.Slot{
-		SlotInfo: &snap.SlotInfo{
-			Snap:      &snap.Info{SuggestedName: "core", Type: snap.TypeOS},
-			Name:      "io-ports-control",
-			Interface: "io-ports-control",
-		},
-	},
-	plug: &interfaces.Plug{
-		PlugInfo: &snap.PlugInfo{
-			Snap:      &snap.Info{SuggestedName: "other"},
-			Name:      "io-ports-control",
-			Interface: "io-ports-control",
-		},
-	},
+var _ = Suite(&IioPortsControlInterfaceSuite{
+	iface: &builtin.IioPortsControlInterface{},
 })
 
-func (s *IoPortsControlInterfaceSuite) TestName(c *C) {
+func (s *IioPortsControlInterfaceSuite) SetUpTest(c *C) {
+	// Mock for OS Snap
+	osSnapInfo := snaptest.MockInfo(c, `
+name: ubuntu-core
+type: os
+slots:
+  test-io-ports:
+    interface: io-ports-control
+`, nil)
+	s.slot = &interfaces.Slot{SlotInfo: osSnapInfo.Slots["test-io-ports"]}
+
+	// Snap Consumers
+	consumingSnapInfo := snaptest.MockInfo(c, `
+name: client-snap
+plugs:
+  plug-for-io-ports:
+    interface: io-ports-control
```

Comment:
This top-level plugs declaration is not needed since you use `plugs: [io-ports-control]` directly down below. I suggest just removing it (or use `plugs: [plug-for-io-ports]` down below).